### PR TITLE
[ISSUE #5235]add aop lock annotation

### PR DESCRIPTION
**Pull request by 985492783**
relate #5235

### CODE REVIEW

**Comment by mrniko** [comment]
Thanks for proposed changes.

**Review by mrniko — CHANGES_REQUESTED** [review]
Why do you mention fair lock and use spin lock? 
Can you introduce Lock type parameter? 
Test is is also required.

**Comment by 985492783** [comment]
> Why do you mention fair lock and use spin lock? Can you introduce Lock type parameter? Test is is also required.

I try to contribute to redisson first. If you think it is a good issue, i will do it more clearer and add unit test

**Comment by 985492783** [comment]
> Why do you mention fair lock and use spin lock? Can you introduce Lock type parameter? Test is is also required.

fix this error and add some unit test

**Comment by 985492783** [comment]
@mrniko review requested

File: redisson-spring-boot-starter/src/main/java/org/redisson/spring/type/LockTypeEnum.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+package org.redisson.spring.type;
+
+/**
+ * @author 985492783@qq.com
+ * @description LockTypeEnum to choose lock.
+ * @date 2023/8/10 15:18
+ */
+public enum LockTypeEnum {
+    SIMPLE,
```
Comment (mrniko) [inline redisson-spring-boot-starter/src/main/java/org/redisson/spring/type/LockTypeEnum.java:24]: Add comments per value

File: redisson-spring-boot-starter/src/main/java/org/redisson/spring/annotation/RMutexLock.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD})
+public @interface RMutexLock {
+    /**
+     * empty placeholder.
+     */
+    String placeholder = "@@empty";
+
+    String name() default placeholder;
```
Comment (mrniko) [inline redisson-spring-boot-starter/src/main/java/org/redisson/spring/annotation/RMutexLock.java:39]: Add comments per setting

**Review by mrniko — CHANGES_REQUESTED** [review]

**Comment by 985492783** [comment]
@mrniko i have solved it

**Review by mrniko — CHANGES_REQUESTED** [review]
lock classes should be moved to redisson\src\main\java\org\redisson\spring\lock folder

**Comment by 985492783** [comment]
> lock classes should be moved to redisson\src\main\java\org\redisson\spring\lock folder

@mrniko done

**Comment by mrniko** [comment]
Sorry, but it's still in org.redisson.api package

**Pull request closed without merging**